src/api/wrap-test-function.js

```diff
@@ -9,6 +9,11 @@ export default function wrapTestFunction (fn) {
         const errList    = new TestCafeErrorList();
         const markeredfn = testRunTracker.addTrackingMarkerToFunction(testRun.id, fn);
 
+        testRun.observedCallsites = {
```

Comment:
I feel it will be better to introduce a new class `ObservedCallsiteStorage`, that will have the `clear` method that we should call here.